client/src/features/edit/edit.ts

```diff
@@ -0,0 +1,253 @@
+import { Action } from "../../base/actions/action"
+import { Command, CommandExecutionContext, CommandResult } from "../../base/commands/command"
+import { SModelElement, SModelRoot } from "../../base/model/smodel"
+import { Editable, hasEditFeature, isEditable } from "./model"
+import { injectable } from "inversify"
+import { IVNodeDecorator } from "../../base/views/vnode-decorators"
+import { VNode } from "snabbdom/vnode"
+import { setClass } from "../../base/views/vnode-utils"
+import { SControlPoint, SEdge } from "../../graph/sgraph"
+import { centerOfLine, Point } from "../../utils/geometry"
+import { SelectAction } from "../select/select"
+import { findParent } from "../../base/model/smodel-utils"
+import { ElementMove } from "../move/move"
+import { KeyListener } from "../../base/views/key-tool"
+
+export class ShowControlPointsAction implements Action {
+    kind: string = ShowControlPointsCommand.KIND
+
+    constructor(public priviousAction?: Action) {
+    }
+}
+
+export class ShowControlPointsCommand implements Command {
+    static KIND: string = "controlPointsVisible"
+
+    constructor(public action: ShowControlPointsAction) {
+    }
+
+    execute(context: CommandExecutionContext): CommandResult {
+        const sModelRoot = context.root
+
+        const createControlPoint = (cptype: string, id: string, position: Point) => {
```

Comment:
Please convert to protected method. It's much easier to reuse/customize for the users